logictest: enable local-mixed-25.3 config

shailendra-patel opened the pull request:
**bootstrap: add 25.3 bootstrap data**
Obtained by running (on release-25.3 branch):
```
 ./dev build sql-bootstrap-data && bin/sql-bootstrap-data
```
Epic: none
Release note: None
Fixes: [REL-2735](https://cockroachlabs.atlassian.net/browse/REL-2735)

shailendra-patel committed:
bootstrap: add 25.3 bootstrap data

Obtained by running (on release-25.3 branch):
```
./dev build sql-bootstrap-data && bin/sql-bootstrap-data
```

Epic: none

Release note: None

shailendra-patel committed:
logictest: enable local-mixed-25.3 config

Added `local-mixed-25.3` logictests.

Epic: none

Release note: None

shailendra-patel committed:
logictest: update logictests data

Update logictest testdata to skip a statement in mixed version
scenarios.

Epic: none

Release note: None

michae2 reviewed (commented):
Do we need to add `local-mixed-25.3` to the skipif flags in so many places? Maybe removing `DisableSchemaLockedByDefault` will allow us to remove some of them?

> Reviewed 6 of 6 files at r1, 11 of 11 files at r2, 4 of 16 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-:-OWats078B4Maxzkt00U:bhfh4ee)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained

___
*[`pkg/sql/logictest/testdata/logic_test/udf_in_table` line 78 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-OWav-j9E4ZcPCRcHQP5:-OWav-j9E4ZcPCRcHQP6:bru51yn) ([raw file](https://github.com/cockroachdb/cockroach/blob/c499c01fee0d72db096ba6777aa6d068ddd6b3ed/pkg/sql/logictest/testdata/logic_test/udf_in_table#L78)):*
> ```
> # Make sure that column DEFAULT and ON UPDATE expressions are properly serialized
> # and deserialized.
> skipif config local-mixed-25.2 
> ```

Would be nice to trim these trailing spaces here (and in other skipif lines).
___
*[`pkg/sql/logictest/logictestbase/logictestbase.go` line 514 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-OWasAq-1PyTLY8CJK_G:-OWasAq-1PyTLY8CJK_H:bsl41xx) ([raw file](https://github.com/cockroachdb/cockroach/blob/c499c01fee0d72db096ba6777aa6d068ddd6b3ed/pkg/sql/logictest/logictestbase/logictestbase.go#L514)):*
> ```Go
> 		DisableUpgrade:               true,
> 		DeclarativeCorpusCollection:  true,
> 		DisableSchemaLockedByDefault: true,
> ```

@fqazi do you know if we can remove this `DisableSchemaLockedByDefault` setting in local-mixed-25.3?


<!-- Sent from Reviewable.io -->

fqazi commented:
@michae2 We should be able to remove the disable flag. Even though it's not enabled by default on that release, the logic tests all had it enabled

shailendra-patel commented:
Thank you for the review @michae2 @fqazi. As per suggestion I have removed `DisableSchemaLockedByDefault`. Also removed the unnecessary `skipif` logic for `local-mixed-25.3`. For following files i have to add `skipif` otherwise few unit tests fails in CI:
```
pkg/sql/logictest/testdata/logic_test/crdb_internal_catalog
pkg/sql/logictest/testdata/logic_test/owner
pkg/sql/logictest/testdata/logic_test/row_level_security
```

PTAL!

michae2 reviewed (commented):
Could you push a revision with all the `local-mixed-25.3` skips removed, so that we can take a look at the test failures and propose fixes for them?

> Reviewed 16 of 16 files at r4, 14 of 14 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-:-OWr2eioA_dR5pfSFh10:b-63xi56)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @shailendra-patel)



<!-- Sent from Reviewable.io -->

shailendra-patel committed:
logictest: update logictests data

Remove `skipif` from logictest testdata for local-mixed-25.3.

Epic: none

Release note: None

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (once the SQL team is ok with the test changes)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-:-OWt6QQS4iRvwuICOPJ2:bc5ymfp)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->

celiala commented:
listing [test failures from unit test run](https://mesolite.cluster.engflow.com/invocations/default/49adce44-9793-44be-a88a-ce3be78bd5ee
):

in pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test:
- TestLogic_crdb_internal_catalog [1]
- TestLogic_row_level_security/alter_policy [2]
- TestLogic_owner [3]

[1] TestLogic_crdb_internal_catalog -- for this, let's try to rewrite the test?

e.g. `dev test pkg/sql/logictest/tests/local --rewrite -f TestLogic_crdb_internal_catalog`

[2] TestLogic_row_level_security/alter_policy
```
=== RUN   TestLogic_row_level_security/alter_policy
    logic.go:2336: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test_/local-mixed-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/row_level_security:1706: 
        expected success, but found
        (42501) new row violates row-level security policy for table "alter_policy_table"
        errors.go:323: in CheckFailed()
--- FAIL: TestLogic_row_level_security/alter_policy (0.55s)
```

[3] TestLogic_owner
```
=== RUN   TestLogic_owner
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestLogic_owner1933371066
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[05:22:47] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test_/local-mixed-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/owner: 48 statements
    logic.go:3198: let $testuser_job_id = 1095437405158735873
    logic.go:3198: let $testuser2_job_id = 1095437409852817409
    logic.go:3198: let $node_job_id = 100
[05:22:47] --- done: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test_/local-mixed-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/owner with config local-mixed-25.3: 58 tests, 0 failures
    logic.go:4365: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test_/local-mixed-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/owner:275: error while processing
    logic.go:4365: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-25.3/local-mixed-25_3_test_/local-mixed-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/owner:275: 
        expected success, but found
        (0A000) ALTER JOB OWNER requires version 1000025.1
        alter_job_owner.go:64: in startExec()
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_owner1933371066
--- FAIL: TestLogic_owner (2.63s)

```

celiala commented:
for TestLogic_crdb_internal_catalog, rewriting the test did nothing. But [this](https://cockroachlabs.atlassian.net/wiki/spaces/RE/pages/4403789867/pkg+clusterversion+how+to+fix+expected+test+failures#TestLogic_crdb_internal_catalog
) should do the trick, i.e.:

<img width="716" height="530" alt="image" src="https://github.com/user-attachments/assets/2c515cee-9b91-4725-b693-215fa42d3d1e" />

shailendra-patel commented:
> @spilchen does something like this look right?
> 
> ```diff
> diff --git a/pkg/sql/logictest/testdata/logic_test/row_level_security b/pkg/sql/logictest/testdata/logic_test/row_level_security
> index 464c48354e2..6dd63bbfc74 100644
> --- a/pkg/sql/logictest/testdata/logic_test/row_level_security
> +++ b/pkg/sql/logictest/testdata/logic_test/row_level_security
> @@ -1696,48 +1696,60 @@ CREATE POLICY p ON alter_policy_table FOR INSERT WITH CHECK (false);
>  statement error pq: new row violates row-level security policy for table "alter_policy_table"
>  INSERT INTO alter_policy_table VALUES (1, 'one'), (2, 'two'), (3, 'three');
>  
> +skipif config local-mixed-25.3
>  statement error pq: only WITH CHECK expression allowed for INSERT
>  ALTER POLICY p ON alter_policy_table USING (true);
>  
> +skipif config local-mixed-25.3
>  statement ok
>  ALTER POLICY p ON alter_policy_table WITH CHECK (nextval('seq1') < 10000);
>  
> +skipif config local-mixed-25.3
>  statement ok
>  INSERT INTO alter_policy_table VALUES (1, 'one'), (2, 'two'), (3, 'three');
>  
> +skipif config local-mixed-25.3
>  query I
>  SELECT c1 FROM alter_policy_table ORDER BY c1;
>  ----
>  
> +skipif config local-mixed-25.3
>  statement ok
>  ALTER POLICY p ON alter_policy_table RENAME TO p_ins;
>  
> +skipif config local-mixed-25.3
>  statement ok
>  CREATE POLICY p ON alter_policy_table FOR SELECT TO aux1 USING (c1 > 0);
>  
> +skipif config local-mixed-25.3
>  query I
>  SELECT c1 FROM alter_policy_table ORDER BY c1;
>  ----
>  
> +skipif config local-mixed-25.3
>  statement error pq: policy "p_sel" for table "alter_policy_table" does not exist
>  ALTER POLICY p_sel ON alter_policy_table WITH CHECK (true);
>  
> +skipif config local-mixed-25.3
>  statement error pq: WITH CHECK cannot be applied to SELECT or DELETE
>  ALTER POLICY p ON alter_policy_table WITH CHECK (true);
>  
> +skipif config local-mixed-25.3
>  statement ok
>  ALTER POLICY p ON alter_policy_table TO alter_policy_role,aux1,aux2 USING (c1 != 1);
>  
> +skipif config local-mixed-25.3
>  query I
>  SELECT c1 FROM alter_policy_table ORDER BY c1;
>  ----
>  2
>  3
>  
> +skipif config local-mixed-25.3
>  statement ok
>  ALTER POLICY p ON alter_policy_table RENAME TO p_sel;
>  
> -skipif config schema-locked-disabled
> +skipif config schema-locked-disabled local-mixed-25.3
>  query TT
>  SHOW CREATE TABLE alter_policy_table;
>  ----
> @@ -1765,6 +1777,7 @@ alter_policy_table  CREATE TABLE public.alter_policy_table (
>                      CREATE POLICY p_ins ON public.alter_policy_table AS PERMISSIVE FOR INSERT TO public WITH CHECK (nextval('public.seq1'::REGCLASS) < 10000:::INT8);
>                      CREATE POLICY p_sel ON public.alter_policy_table AS PERMISSIVE FOR SELECT TO aux1, alter_policy_role, aux2 USING (c1 != 1:::INT8);
>  
> +skipif config local-mixed-25.3
>  query TTTTTT colnames
>  SELECT name,cmd,type,roles,using_expr,with_check_expr
>  FROM [SHOW POLICIES FOR alter_policy_table]
> @@ -1774,12 +1787,15 @@ name   cmd     type        roles                          using_expr      with_c
>  p_sel  SELECT  permissive  {alter_policy_role,aux1,aux2}  c1 != 1:::INT8  ·
>  p_ins  INSERT  permissive  {public}                       ·               nextval('public.seq1'::REGCLASS) < 10000:::INT8
>  
> +skipif config local-mixed-25.3
>  statement ok
>  SET ROLE root;
>  
> +skipif config local-mixed-25.3
>  statement error pq: cannot drop sequence seq1 because other objects depend on it
>  DROP SEQUENCE seq1;
>  
> +skipif config local-mixed-25.3
>  # Change the policy so there isn't a dependency on seq1 anymore.
>  statement ok
>  ALTER POLICY p_ins ON alter_policy_table WITH CHECK (true);
> ```

@michae2 @spilchen should i add the above suggested changes?

michae2 commented:
> @michae2 @spilchen should i add the above suggested changes?

@shailendra-patel yes, let's try it out and see how it looks.

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for giving it a shot. I think we can merge this, then.

> Reviewed 1 of 4 files at r6, 4 of 4 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/151115#-:-OX4Omd95cyEznclkh8w:b-2hics9)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @shailendra-patel)



<!-- Sent from Reviewable.io -->

shailendra-patel commented:
TFTR @michae2 @RaduBerinde

bors r=michae2,RaduBerinde

shailendra-patel commented:
Retry build failure.

bors r+
